src/relay/transforms/partition_graph.cc

```diff
@@ -374,19 +270,16 @@ class Partitioner : public ExprMutator {
    * node to create the function for the subgraph.
    */
   void CreateFunction(AnnotatedRegion region, const CallNode* call) {
-    // Create fields which is a unique list of outputs. Also populate
-    // region_return_indices_ map which maps parent of compiler_end node to
-    // corresponding index in fields.
+    // Create fields which is a unique list of outputs.
     Array<Expr> fields;
     int i = 0;
     for (auto ret : region->GetOutputs()) {
       auto ret_node = Downcast<Call>(ret)->args[0];
       // Don't duplicate outputs.
-      if (!region_return_indices_.count(region) ||
-          !region_return_indices_[region].count(ret_node)) {
-        auto ret_expr = VisitExpr(ret_node);
+      if (!region_func_meta_[region].out_expr_indices.count(ret_node)) {
+        auto ret_expr = MixedModeMutator::VisitExpr(ret_node);
         fields.push_back(ret_expr);
-        region_return_indices_[region][ret_node] = i;
+        region_func_meta_[region].out_expr_indices[ret_node] = {i, TupleGetItem()};
```

Comment:
Can we move this at the end of this function so that we can properly initialize `TupleGetItem` with `func_call`? This way `TupleGetItem` is always defined so you can clean up `GetFunctionOutput` as well.

Reply:
Good point. Modified accordingly.